### PR TITLE
core: contractCall from native contracts

### LINKED ISSUE #1200: Make sure we can make calls from native contracts
Evalute effects of neo-project/neo#1700 for our code.

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Closes #1200

I made an example of such call, but not sure if this testcase covers the problem. If so, than we are able to perform calls from native contracts and it might be necessary to implement a separate method for such calls.

### CODE REVIEW

**Comment by roman-khimov** [comment]
This is native contract calling another native contract. Have you tried calling regular contracts from native contract?

**Comment by roman-khimov** [comment]
And it's spawning another VM, which doesn't seem to be correct.

**Comment by roman-khimov** [comment]
At the same time C# node mostly changes `ApplicationEngine` here which is an internal (to that node's implementation) thing. Still, we should consider this scenario without new VM spawning, it should be possible to do using our current VM/interop.Context APIs.

**Comment by AnnaShaleva** [comment]
Now we have VM into interop context, so we're able to call other contracts from native contracts, but it's strange a bit, because if VM fails during invoking inner `contractCall`, we'll have failed VM after inner `cantractCall` completed. Consequently, outer `contractCall` fails, as these calls share the VM.

**Comment by roman-khimov** [comment]
> if VM fails during invoking inner contractCall, we'll have failed VM after inner cantractCall completed. Consequently, outer contractCall fails, as these calls share the VM.

Well, that's a feature. That's what would happen with regular contracts calling each other and native contracts shouldn't be that much different.

File: pkg/core/native_contract_test.go [hunk 1]
```diff
@@ -317,7 +317,7 @@ func TestNativeContract_InvokeOtherContract(t *testing.T) {
 	})
 
 	t.Run("non-native contract", func(t *testing.T) {
-		// put some other contract into chain
+		// put some other contract into chain (this contract just pushes `5` on stack)
```
Comment (roman-khimov) [inline pkg/core/native_contract_test.go:320]: This belongs to some different commit.

File: pkg/core/interop/context.go [hunk 2]
```diff
@@ -35,7 +35,6 @@ type Context struct {
 	Notifications []state.NotificationEvent
 	Log           *zap.Logger
 	Invocations   map[util.Uint160]int
-	ScriptGetter  vm.ScriptHashGetter
```
Comment (roman-khimov) [inline pkg/core/interop/context.go:38]: We probably no longer need `vm.ScriptHashGetter` type either.

File: pkg/core/native_contract_test.go [hunk 3]
```diff
@@ -95,6 +126,38 @@ func (tn *testNative) sum(_ *interop.Context, args []stackitem.Item) stackitem.I
...
+	vm.Estack().PushVal(stackitem.NewArray([]stackitem.Item{})) // no args
+	vm.Estack().PushVal(args[1])                                // method
+	vm.Estack().PushVal(args[0])                                // contract hash
+	err := contractCall(ic)
+	if err != nil {
+		return stackitem.NewBigInteger(big.NewInt(-1))
+	}
+	_ = vm.Run()
+	if vm.HasFailed() {
+		return stackitem.NewBigInteger(big.NewInt(-2))
```
Comment (roman-khimov) [inline pkg/core/native_contract_test.go:140]: It's a bit strange that we're returning something here, but I guess there are not that many options we have.

Comment (AnnaShaleva) [reply pkg/core/native_contract_test.go:140]: It was a test method, so I just put something there. What should we do in this case?

Comment (roman-khimov) [reply pkg/core/native_contract_test.go:140]: Well, the vm already has failed. We shouldn't touch it, but at the same time we have to return something for `Func`, so it probably could be just about anything.
